### PR TITLE
Liberty tools deep dive

### PR DESCRIPTION
**Pull request by awisniew90**

### CODE REVIEW

**Comment by awisniew90** [comment]
Issue #2958

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Prefer another IDE? Liberty Tools for Eclipse IDE maintains functional parity with plugins for link:https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Visual Studio Code] and link:https://plugins.jetbrains.com/plugin/14856-liberty-tools/[Intellij IDEA].
+
+== Requirements
+
+Liberty Tools for Eclipse IDE is designed to work with link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#software-requirements[supported versions] of both 
+the IDE for Java EE Developers and the IDE for Java Developers. While Java 17 is needed to run the tool, you can choose a different version of Java to run your application. 
+
+Since Liberty Tools uses the Liberty Maven and Liberty Gradle plugins behind the scenes, these plugins must be configured in your applications `pom.xml` or `build.gradle` files. Liberty Tools will also need to run the Maven or Gradle CLI which can be accomplished by configuring the Maven and Gradle home in the link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#setting-preferences[tool's preferences], or including a wrapper with the application.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: applications `pom.xml` => application's `pom.xml`

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Keeping in line with the app-centric design, Liberty Tools provides code assistance for both Jakarta EE and MicroProfile APIs. Code completion, hover descriptions, diagnostics and quick-fixes save time, reduce errors, and keep you focused on your code. Liberty Tools accomplishes this by integrating with the link:https://github.com/eclipse/lsp4jakarta[Language Server for Jakarta EE] and the link:https://github.com/eclipse/lsp4mp[Language Server for MicroProfile], both open source projects which are continually adding new features and support.
+
+While much of the runtime mechanics are automated with Liberty Tools, some runtime configuration is, of course, necessary. However, this process has been streamlined with config assistance for `server.xml`, `bootstrap.properties`, and `server.env` files. Rather than having to refer to the documentation to find the correct Liberty feature to add or the appropriate value to set for your specific logging attribute, Liberty Tools provides lists of options and descriptions all without having to leave Eclipse.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc:33]: to set for your specific logging attribute
  =>
to set for your specific logging attribute, (for example),...

---
Maybe??  Getting right into logging seems maybe overly specific.. I get what you're saying but perhaps clearer adding that?

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 1 above)
Comment (scottkurz) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: or Gradle CLI  -> or Gradle executable?

Is that better?  Eh, possibly..take it or leave it.

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+the IDE for Java EE Developers and the IDE for Java Developers. While Java 17 is needed to run the tool, you can choose a different version of Java to run your application. 
+
+Since Liberty Tools uses the Liberty Maven and Liberty Gradle plugins behind the scenes, these plugins must be configured in your applications `pom.xml` or `build.gradle` files. Liberty Tools will also need to run the Maven or Gradle CLI which can be accomplished by configuring the Maven and Gradle home in the link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#setting-preferences[tool's preferences], or including a wrapper with the application.
+
+== Managing your applications
+
+Any application in your Eclipse workspace that is configured with the Liberty Maven or Liberty Gradle plugin is automatically enabled to run with Liberty Tools. Actions are available to manage these apps such as `Start`, `Stop`, `Debug` and `Run Tests`, and you can run these actions from either the Liberty Dashboard or the Eclipse "Run As" menu.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: > Any application in your Eclipse workspace that is configured with the Liberty Maven or Liberty Gradle plugin is automatically enabled to run with Liberty Tools.  

 
Not quite..it's actually projects with a src/main/liberty/config/server.xml (that's close enough...though technically, one of the other two config files)

Comment (scottkurz) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: "from either the Liberty Dashboard or the Eclipse "Run As" menu."

I like to speak of the three options with the third being the Run/Debug configurations menu + toolbar.. but your rendition might be better , accurate enough and more succint..maybe the other way is implied.

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+=== Liberty Dashboard
+
+The Liberty Dashboard consolidates the Liberty enabled applications in your workspace into a single view. To open the dashboard, click on Open Liberty UFO icon on the top tray. To run an action, simply right click on an app in the list.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: on the top tray (or press Alt+L) 

Promote the keyboard shortcut ?

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+=== Eclipse Run As menu
+
+Liberty Tools offers the same set of actions in the Eclipse "Run As" menu that appears when you right click on a project in the Project Explorer view.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-runas-menu.png[Run As menu,width=70%,align="center"]
+
+=== Starting your app
+
+To start your application in dev mode, select either the `Start` or `Start...` actions. The latter will open a `Run Configurations` dialog box where you can add link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#additional-parameters[additional parameters] to dev mode or configure the JRE to use when running the app. When a start action is selected, Liberty Tools opens a `Terminal` tab and starts dev mode.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: Nice to mention the dev parameters but you can add anything..what about:

"you can add to the command-line parameters (e.g. .additional parameters to dev mode

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+To start your application in dev mode, select either the `Start` or `Start...` actions. The latter will open a `Run Configurations` dialog box where you can add link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#additional-parameters[additional parameters] to dev mode or configure the JRE to use when running the app. When a start action is selected, Liberty Tools opens a `Terminal` tab and starts dev mode. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-run-config.gif[Run Configurations,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-terminal-start.png[Terminal Start,width=70%,align="center"]
+
+NOTE: The `Start in container` action will link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#devc-container-mode[start dev mode in a local Docker container].
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: Maybe link to the PATH setup?

..and requires 'docker' on the PATH? (link to
https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#docker

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+NOTE: The `Start in container` action will link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#devc-container-mode[start dev mode in a local Docker container]. 
+
+=== Running tests
+
+Tests are important for any application. Liberty Tools makes it easy to run your unit and integration tests by either selecting the `Run tests` action, or simply clicking `Enter` in the terminal. You can even view the results of your latest run by selecting `View test reports`.
+
+=== Debugging your app
+
+The Liberty runtime is configured with port 7777 for debugging by default. However, this requires that port to be available on your system and for you to manually attach a debugger. With Liberty Tools, you can start your app using the "Debug" action which will find an available port and automatically attach the Eclipse debugger to it.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc:80]: Could the extra background info be a bit confusing?  You could just include the last sentence?

```suggestion
With Liberty Tools, you can start your app using the "Debug" action which will find an available port and automatically attach the Eclipse debugger to it. 
```

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+=== Debugging your app
+
+The Liberty runtime is configured with port 7777 for debugging by default. However, this requires that port to be available on your system and for you to manually attach a debugger. With Liberty Tools, you can start your app using the "Debug" action which will find an available port and automatically attach the Eclipse debugger to it. 
+
+== Get coding!
+
+=== Jakarta EE 
+
+Code completion for Jakarta EE makes it easy to add classes and methods to your app. Just type `CTRL` + `SPACE` from within any Java file to see a list of available code snippets to choose from. Additionally, diagnostics flag errors and suggest quick-fixes to implement on the spot.
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: I feel like I usually see this as `Ctrl + Space`  or `CTRL +Space`  but don't usually see SPACE in caps.

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+Liberty Tools streamlines the process of configuring the Liberty runtime through config completion for `server.xml`, `bootstrap.properties` and `server.env` files. Type `CTRL` + `SPACE` from within these files to get lists of suggested configuration elements, properties, and values to add. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-server-config.gif[Server config code completion,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-bootstrap-props.gif[Bootstrap properties code completion,width=70%,align="center"]
+
+== Learn more
```
Comment (scottkurz) [inline posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: Is this supposed to be empty?

**Review by scottkurz — COMMENTED** [review]
Nice job on the animations.   They look good and pack a lot of info into a small area.

Some of these comments you can take or leave.

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 2 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc:33]: I can change "specific logging attribute" to something else.  I was trying to paint the picture of "a tedious task", not necessarily pointing out anything in particular.

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 4 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: Good idea

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 5 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: nice

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 6 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: done

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 7 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc:80]: hmm.... i like it in there to emphasize what we are simplifying with the tool. Plus someone might read just the last line and think "doesnt Liberty already do this?"

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 8 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: ok - changed

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 9 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: No!   Thanks for the catch.  Changing this to a "Feedback" section

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 3 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: Yea I was originally thinking of explaining the shortcut keys too, but settled on talking to just the two ways

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 1 above)
Comment (awisniew90) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc]: changing to executable

File: posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc (hunk 7 above)
Comment (scottkurz) [reply posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc:80]: OK that's fine.

**Review by scottkurz — APPROVED** [review]

**Pull request merged**